distributed/bokeh/scheduler.py

```diff
@@ -933,6 +933,9 @@ def __init__(self, scheduler, width=800, **kwargs):
         self.names = ['worker', 'ncores', 'cpu', 'memory', 'memory_limit',
                       'memory_percent', 'num_fds', 'read_bytes', 'write_bytes',
                       'cpu_fraction']
+        workers = self.scheduler.workers.values()
+        self.custom_metrics_names = list(workers[0].info.get('custom_metrics_names', []))
```

Comment:
This is the way I managed to get the custom_metrics_names, but maybe there is a better way to pass them around?

Reply:
What if only some workers have some metrics.  I can imagine this being the case with a cluster that has some GPU workers and some non-GPU workers for example.

I'll suggest the following:

    names = sorted(sum([w.info.get('custom_metrics', []) for w in workers], []))

Reply:
Good point, I didn't think of that. I'll add a test for this.

Reply:
Looks like I forgot to form it into a set

    names = sorted({n for w in workers for n in w.info.get('custom_metrics', [])})

Presumably there is also dict.get call later on when populating the tables.